app/src/main/kotlin/com/simplemobiletools/flashlight/helpers/MyCameraImpl.kt

```diff
@@ -14,7 +15,7 @@ import com.simplemobiletools.flashlight.extensions.updateWidgets
 import com.simplemobiletools.flashlight.models.Events
 import org.greenrobot.eventbus.EventBus
 
-class MyCameraImpl(val context: Context) {
+class MyCameraImpl private constructor(val context: Context, private val cameraTorchListener: CameraTorchListener?) {
```

Comment:
Hi @tibbi. Just a suggestion.  
The implementation in this class is a bit confusing because we have two implementations for Nougat+ and PreNougat devices. Would it be worth it to refactor this class to share the same API on both implementations (create a common Interface for both)

Reply:
if it works as it is, lets leave it for now

Reply:
Hi, 
I had to separate the implementations into different files `LollipopFlash` and `MashmallowPlusFlash` for simplicity. It became hard to manage states without having a centralized place.